pkg/workspace/catalog.go

```diff
@@ -18,8 +19,11 @@ import (
 
 const (
 	magicModuleFilename = "_module.json"
+	CatalogNameFormat   = `^([A-Za-z0-9][-A-Za-z0-9_.]{0,62})?$`
```

Comment:
Seems a little aggro.  This doesn't bear a ton of load, either, because it's really only a local name.  It doesn't get serialized and shipped around in other key persistent data structures.

That said: it's easy to loosen this up later, than to lock it down later, so... :+1:

Reply:
Hang on, what's the `?` for?

Also do you need the parens for a capture grouping?  I think not, because you're only using it to check for a match overall, not to extract things, yes?

Reply:
The `()?` allows the validator to accept an empty string. We _could_ remove it because we are handling the empty string explicitly. But it's not a huge deal either way.